source/robot_model/src/Model.cpp

```diff
@@ -150,6 +150,45 @@ state_representation::Jacobian Model::compute_jacobian(const state_representatio
   return this->compute_jacobian(joint_positions, frame_id);
 }
 
+Eigen::MatrixXd Model::compute_jacobian_time_derivative(const state_representation::JointPositions& joint_positions,
+                                                        const state_representation::JointVelocities& joint_velocities,
+                                                        unsigned int frame_id) {
+  if (joint_positions.get_size() != this->get_number_of_joints()) {
+    throw (exceptions::InvalidJointStateSizeException(joint_positions.get_size(), this->get_number_of_joints()));
+  }
+  // compute the jacobian from the joint state
+  pinocchio::Data::Matrix6x dJ(6, this->get_number_of_joints());
+  dJ.setZero();
```

Comment:
Is that necessary?

Reply:
maybe not, but is a good practice, no? I don't want to have `nan` values and to be honest I copy the code for computing the jacobian
But if you want I can remove this line.

Reply:
No it's fine I don't have a preference. I was just curious

Reply:
it is actually but you can do that in one line `pinocchio::Data::Matrix6x dJ = Eigen::MatrixXd::Zero(6, this->get_number_of_joints());`